admiral/pkg/controller/common/types_test.go

```diff
@@ -191,3 +191,50 @@ func TestMapRange(t *testing.T) {
 	assert.Equal(t, 3, numOfIter)
 
 }
+
+func TestSidecarEgressGet(t *testing.T) {
+
+	egressMap := NewSidecarEgressMap()
+	egressMap.Put("pkey1", "pkey2", "fqdn", map[string]string{"pkey2": "pkey2"})
+
+	ctx, cancel := context.WithDeadline(context.Background(), time.Now().Add(3*time.Second))
+	defer cancel()
+
+	go func(ctx context.Context) {
```

Comment:
```suggestion
       wg := synced.WaitGroup()
       wg.Add(1)
	go func(ctx context.Context) {
	defer wg.Done()
```
Understand it is test code, but to avoid possible memory leak, waitgroup can help in this case.

Reply:
Thanks @ccfishk . I'm sorry I'm trying to understand where could this lead to a memory leak. As per my understanding, as the context's deadline expires, `ctx.Done()`  would get called and the `go routine` in that case should return. Please correct me if I'm wrong.

Reply:
@shriramsharma  Sorry for the late response. 
I mean goroutine leak, which is memory leak, coz goroutine is not garbage cleaned by runtime object, only if the parent process specifically terminate the  goroutine before it exit. 
One example is that parent process stops for error while goroutine is left as it is. The clean way includes
 - signal (like channle close make(chan interface{})) between parent and the routine,
 -  waitgroup  synchronize

Reply:
@ccfishk , I think I understand where you're getting at. Thanks. The issue is with the unpredictable order in which the `Done()` method will be called. There could be a scenario where the parent go routine exits prior to the child. 
I added `Waitgroup` as you suggested.